uniffi_macros/src/object.rs

```diff
@@ -121,6 +121,7 @@ fn interface_impl(object: &ObjectItem, options: &DeriveOptions) -> TokenStream {
         // if they are not, but unfortunately it fails with an unactionably obscure error message.
         // By asserting the requirement explicitly, we help Rust produce a more scrutable error message
         // and thus help the user debug why the requirement isn't being met.
+        #[cfg(not(target_arch = "wasm32"))]
```

Comment:
Removing this check seems to be removing it for far more things that just objects holding `Future`s?